tensorizer/stream_io.py

```diff
@@ -1,128 +1,77 @@
-# the capacity can be queried and set using the fcntl(2) F_GETPIPE_SZ and
-# F_SETPIPE_SZ operations. See fcntl(2) for more information.
-#
-# =============================================================================
-# From `fcntl(2)` manpage:
-#
-# Changing the capacity of a pipe
-#
-# F_SETPIPE_SZ (int; since Linux 2.6.35)
-# Change the capacity of the pipe referred to by fd to be at least arg bytes.
-# An unprivileged process can adjust the pipe capacity to any value between the
-# system page size and the limit defined in /proc/sys/fs/pipe−max−size
-# (see proc(5)). Attempts to set the pipe capacity below the page size are
-# silently rounded up to the page size. Attempts by an unprivileged process to
-# set the pipe capacity above the limit in /proc/sys/fs/pipe−max−size yield the
-# error EPERM; a privileged process (CAP_SYS_RESOURCE) can override the limit.
-#
-# When allocating the buffer for the pipe, the kernel may use a capacity larger
-# than arg, if that is convenient for the implementation. (In the current
-# implementation, the allocation is the next higher power-of-two page-size
-# multiple of the requested size.) The actual capacity (in bytes) that is set
-# is returned as the function result.
-#
-# Attempting to set the pipe capacity smaller than the amount of buffer space
-# currently used to store data produces the error EBUSY.
-#
-# Note that because of the way the pages of the pipe buffer are employed when
-# data is written to the pipe, the number of bytes that can be written may be
-# less than the nominal size, depending on the size of the writes.
-#
-# F_GETPIPE_SZ (void; since Linux 2.6.35)
-# Return (as the function result) the capacity of the pipe referred to by fd.
-#
-# =============================================================================
-# Constant for `F_SETPIPE_SZ`, as python3's `fcntl` module doesn't have this
-# defined -- despite the documentation saying that they're there.
-#
-# TODO: Make this work or fail gracefully on non-Linux systems. Not sure if
-#       this is really relevant, as I don't even know if CUDA is available on
-#       non-Linux systems in a production sense.
-F_SETPIPE_SZ = 1031
-
-# Read our max-fd-size, fall back to 1mb if invalid.
-PIPE_BUF_SZ = 1024 * 1024
-try:
-    pipe_file = open("/proc/sys/fs/pipe-max-size", "r")
-    pipe_buf_sz = int(pipe_file.read())
-    logger.debug(f"pipe-max-size: {pipe_buf_sz}")
-except IOError as e:
-    logger.warning(
-        f"Could not read /proc/sys/fs/pipe-max-size: {e.strerror}"
-    )
+curl_path = shutil.which("curl")
+
+default_s3_endpoint = "object.ord1.coreweave.com"
+if sys.platform != "win32":
+    _s3_config_paths = (os.path.expanduser("~/.s3cfg"),)
+else:
+    # s3cmd generates its config at a different path on Windows by default,
+    # but it may have been manually placed at ~\.s3cfg instead, so check both.
+    _s3_config_paths = tuple(map(os.path.expanduser,
```

Comment:
Yep, WSL (Ubuntu) reports `sys.platform` as `"linux"` and it will search in `~/.s3cfg` on the WSL installation's filesystem. It treats it as a regular Linux installation so it won't search for external Windows paths in `/mnt/c/` by default, but I implemented `_get_s3cfg_values()` to take `config_paths` as a parameter, so we can expose it in `open_stream()` to let a user add on arbitrary search paths beyond the defaults if it seems like a good idea.